### ISSUE TITLE
DerivedProperty has both detach and dispose functions

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/axon#61]
DerivedProperty is a subtype of Property, which now has a dispose function. Two issues with this:

1. Currently, calling dispose does not call tandem.removeInstance, which is a memory leak.

2. DerivedProperty now has both dispose and detach functions. What is the use case for detach?

**Comment by pixelzoom** [comment]
Here's how I fixed the leak in DerivedProperty.dispose:

```js
      dispose: function() {
        Property.prototype.dispose.call( this );
        this.detach();
      },
```

When instrumented with data collection, I suspect that most (all?) client uses of DerivedProperty.detach will need to be replaced with DerivedProperty.dispose.

@jonathanolson @samreid @jbphet Please comment on why we need both detach and dispose for DerivedProperty.

**Comment by jonathanolson** [comment]
I haven't worked with dispose/detach on DerivedProperty, I'd defer to the others.

**Comment by pixelzoom** [comment]
5/28/15 dev meeting: Consolidate into dispose, delete detach - there is no value to a DerivedProperty that has been detached from its dependency properties.

**Comment by pixelzoom** [comment]
@samreid If you have a contrary opinion about this, please comment ASAP.

**Comment by samreid** [comment]
Sounds good, thanks for keeping me in the loop.

**Comment by pixelzoom** [comment]
This also applied to Multilink, detach should be rename to dispose.

**Comment by pixelzoom** [comment]
Work on this is done. Most of the usages of 'detach' were for scenery nodes, and those were of course not touched.  The number of usages of 'detach' for DerivedProperty and Multilink was very small, and that's a bit disconcerting - either most such objects exist for the lifetime of a sim, or sims are leaking memory.

Assigned to @aaronsamuel137 to review the changes. Thanks!

**Comment by aaronsamuel137** [comment]
Changes look good to me. Closing the issue.
